catalog/resource_sql_table.go

```diff
@@ -332,6 +332,35 @@ func (ti *SqlTableInfo) diff(oldti *SqlTableInfo) ([]string, error) {
 		statements = append(statements, fmt.Sprintf("ALTER %s %s SET TBLPROPERTIES (%s)", typestring, ti.SQLFullName(), ti.serializeProperties()))
 	}
 
+	// Since we marked `column_info` with `force_new`, this should only include the case where the size of the list is unchanged.
+	if !reflect.DeepEqual(ti.ColumnInfos, oldti.ColumnInfos) {
+		for i, ci := range ti.ColumnInfos {
+			oldCi := oldti.ColumnInfos[i]
+			if ci.Name != oldCi.Name {
+				println("name not equal!")
+				statements = append(statements, fmt.Sprintf("ALTER %s %s RENAME COLUMN %s to %s", typestring, ti.SQLFullName(), oldCi.Name, ci.Name))
+			}
+			if ci.Type != oldCi.Type {
+				println("type not equal!")
+
+				statements = append(statements, fmt.Sprintf("ALTER %s %s ALTER COLUMN %s TYPE %s", typestring, ti.SQLFullName(), ci.Name, ci.Type))
```

Comment:
does not look like this syntax is [supported](https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-ddl-alter-table-manage-column.html#alter-column-clause) - most of the time it requires rewriting the full table to change the type of a column